Integrate auth backend

mingjia-wang opened the pull request:
Minor modifications to the admin invite routes, schemas, and util functions due to design and schema changes.

mingjia-wang committed:
Got register working

mingjia-wang committed:
Completed login, register, refreshToken, and users CRUD routes

ZubairSidhu committed:
Create PR for #32

ZubairSidhu committed:
Add user router

ZubairSidhu committed:
Fix order

ZubairSidhu committed:
Add route to assign segment to user

ZubairSidhu committed:
Return new doc when adding segment

ZubairSidhu committed:
Remove console logs

mingjia-wang committed:
Changes to nodemailer.router and email sendingfunctionality

ZubairSidhu committed:
Add route to get all segments

ZubairSidhu committed:
Enable virtuals for segment schema

ZubairSidhu committed:
Populate segments when getting user data

ZubairSidhu committed:
Enable virtuals for user schema

mingjia-wang committed:
Committing while my code still compiles

mingjia-wang committed:
Finished entire adminInvite flow

mingjia-wang committed:
Completed auth for good

ZubairSidhu committed:
Minor route change

mingjia-wang committed:
Merged dev into branch and resolved conflicts

mingjia-wang committed:
Merge branch 'dev' of https://github.com/ctc-uci/oc-habitats-backend into integrate-auth-backend

mingjia-wang committed:
Merge branch 'integrate-auth-backend' of https://github.com/ctc-uci/oc-habitats-backend into integrate-auth-backend

mingjia-wang committed:
Committing random change git says I made

mingjia-wang committed:
Got confirmation & error messages working

mingjia-wang committed:
Merge branch 'dev' into integrate-auth-backend

ctc-devops commented on services/user.service.js:
```diff
@@ -24,6 +24,7 @@ const updateProfile = async (profileId, updatedProfile) => {
   );
 };
 
+
 const assignSegment = async (userId, segmentId) => {
```
🚫 **[eslint]** <[prettier/prettier](https://github.com/prettier/eslint-plugin-prettier#options)> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Delete `⏎`
```suggestion
const assignSegment = async (userId, segmentId) => {
```

mingjia-wang committed:
Update services/user.service.js

Committing code change suggestion from ReviewDog -- removed extra enter keystroke from line 27 in user.service.js

Co-authored-by: ctc-devops <90984711+ctc-devops@users.noreply.github.com>

mingjia-wang committed:
Added error message to Forgot Password page if email could not be found

mingjia-wang committed:
Merge branch 'integrate-auth-backend' of https://github.com/ctc-uci/oc-habitats-backend into integrate-auth-backend

Needed to pull changes in order to push modifications to Forgot Password functionality

ctian1 commented on app.js:
```diff
@@ -40,11 +40,17 @@ app.use('/auth', authRouter);
+app.use('/users', userRouter);
+app.use('/auth', authRouter);
+// TODO remove
+app.use('/test', [verifyToken, userRouter]);
```
do we need this still?

mingjia-wang committed:
Removed test route from app.js
